src/consumers/reward.processor.ts

```diff
@@ -0,0 +1,11 @@
+import { Processor, Process, OnQueueCompleted, OnQueueError } from '@nestjs/bull';
+import { Job } from 'bull';
+import { Trigger } from 'src/hook/enums/trigger.enum';
+
+@Processor('hooksQueue')
+export class RewardProcessor {
+  @Process(Trigger.REWARD_GRANTED)
+  async grantReward(job: Job<unknown>) {
```

Comment:
It does not grant the reward, it makes further actions when a reward is granted (run hooks registered with trigger "REWARD_GRANTED").